allennlp/tests/models/bidirectional_lm_test.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import torch
+
+from allennlp.common.testing import ModelTestCase
+from allennlp.data.fields import TextField
+from allennlp.data.instance import Instance
+from allennlp.data.iterators import BasicIterator
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenCharactersIndexer
+from allennlp.data.tokenizers import WordTokenizer
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.models.bidirectional_lm import BidirectionalLanguageModel
+from allennlp.modules.seq2seq_encoders import PytorchSeq2SeqWrapper
+from allennlp.modules.text_field_embedders import BasicTextFieldEmbedder
+from allennlp.modules.token_embedders import TokenCharactersEncoder, Embedding
+from allennlp.modules.seq2vec_encoders.cnn_highway_encoder import CnnHighwayEncoder
+from allennlp.nn.util import get_text_field_mask
+from allennlp.training import Trainer
+
+START_TOKENS = ["<s>"]
+END_TOKENS = ["</s>"]
+
+class TestBidirectionalLM(ModelTestCase):
+    def setUp(self):
+        super().setUp()
+
+        sentences = ["This is the first sentence.", "This is yet another sentence."]
+        token_indexers = {
+                "tokens": SingleIdTokenIndexer(start_tokens=START_TOKENS, end_tokens=END_TOKENS),
+                "token_characters": TokenCharactersIndexer(start_tokens=START_TOKENS, end_tokens=END_TOKENS)
+        }
+        tokenizer = WordTokenizer()
+
+        self.instances = [Instance({"tokens": TextField(tokenizer.tokenize(sentence), token_indexers)})
+                          for sentence in sentences]
+
+
+        tokens_to_add = {
+                "tokens": START_TOKENS + END_TOKENS,
+                "token_characters": [c for token in START_TOKENS + END_TOKENS for c in token]
+        }
+
+        self.vocab = Vocabulary.from_instances(self.instances, tokens_to_add=tokens_to_add)
+
+    def test_lm_can_run(self):
```

Comment:
What does this buy you over our typical model test?  I think it's trying to test individual pieces, but those should already be tested.  It's also testing that things work if you configure them correctly, but that's already done in our typical tests (and done better, because it also checks gradients and other things).